### PR TITLE
virttest.qemu_vm: Add function for disable-legacy and disable-modern

### PR DESCRIPTION
**Pull request by spcui**
Signed-off-by: Shuping Cui <scui@redhat.com>

### CODE REVIEW

**Comment by spcui** [comment]
ID: 1241333

**Comment by CongSmile** [comment]
Also need set the filter in guest side, as not all guests support this param.

File: virttest/qemu_vm.py [hunk 1]
```diff
@@ -1119,6 +1119,18 @@ def add_balloon(devices, devid=None, bus=None, use_old_format=None):
                 dev.set_param("id", devid)
             devices.insert(dev)
 
+        def add_disable_legacy(devices, dev, dev_type):
+            options = devices.execute_qemu("-device %s,?" % dev_type)
+            if "disable-legacy" in options:
+                value = params.get("disable_legacy", "off")
+                dev.set_param("disable-legacy", value)
```
Comment (CongSmile) [inline virttest/qemu_vm.py]: How about if there is no such option "disable-legacy" ?

File: virttest/qemu_vm.py [hunk 2]
```diff
@@ -1958,6 +1976,15 @@ def add_balloon(devices, devid=None, bus=None, use_old_format=None):
...
                         use_old_format=use_ofmt)
 
+        for device in devices:
+            virtio_pci_devices = ["virtio-net-pci", "virtio-blk-pci",
+                                  "virtio-scsi-pci", "virtio-balloon-pci",
+                                  "virtio-serial-pci", "virtio-rng-pci"]
+            dev_type = device.get_param("driver")
+            if dev_type in virtio_pci_devices:
+                add_disable_legacy(devices, device, dev_type)
+                add_disable_modern(devices, device, dev_type)
```
Comment (CongSmile) [inline virttest/qemu_vm.py]: If the param set "disable-modern = off", but there is no such option "disable-modern" in qemu, which means the qemu does not support such option.  
There should give a warn or case failed directly, which may cause the real problem in testing.

Comment (suqinhuang) [inline virttest/qemu_vm.py]: In my opinion, it's ok without the warning.

**Comment by suqinhuang** [comment]
Acked-by: Suqin Huang

**Comment by CongSmile** [comment]
Acked-by: Cong Li <coli@redhat.com>

**Comment by xutian** [comment]
LGTM

merged. thanks @spcui  @CongSmile  and @suqinhuang

**Pull request merged**